### PR TITLE
server/server: avoid test case hanging

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR revises the test case `server::server::tests::test_peer_resolve` to avoid hanging forever, so that even if it fails, it would not block tests.

PTAL @disksing @siddontang @BusyJay

### CODE REVIEW

**Comment by hhkbp2** [comment]
Related to #1895

File: src/server/server.rs [hunk 1]
```diff
@@ -475,7 +476,7 @@ mod tests {
             })
             .unwrap();
 
-        rx.recv().unwrap();
+        assert!(rx.recv_timeout(Duration::from_secs(5)).is_ok());
```
Comment (BusyJay) [inline src/server/server.rs]: If it fails, shouldn't thread at L470 panic? Then tx should be dropped.

**Comment by siddontang** [comment]
Do we know why this case can hang up forever?

**Comment by hhkbp2** [comment]
@siddontang 
> Do we know why this case can hang up forever?

No. It seems that @disksing is busy with other stuff without the time to take a look at it. I need this PR so that it would not block `make test`.

File: src/server/server.rs (hunk 1 above)
Comment (hhkbp2) [inline src/server/server.rs]: The server thread will not panic if it doesn't receive messages. the server shutdown is skipped in test case if this assertion is triggered.

**Comment by siddontang** [comment]
any update? @disksing

**Comment by hhkbp2** [comment]
@siddontang 
The issue is caused by the environment variable `http_proxy`, which is set on the shell startup configuration script.

We could still merge this PR since it adds timeout to the receive operation.

PTAL @disksing @BusyJay

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by siddontang** [comment]
CI failed

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
Please fix conflict @hhkbp2

**Pull request merged**